### PR TITLE
fix(loguru): ensure loguru events are automatically patched

### PR DESCRIPTION
**Pull request by mabdinur**
Resolves: https://github.com/DataDog/dd-trace-py/issues/7988

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/contrib/loguru/patch.py [hunk 1]
```diff
@@ -14,6 +15,8 @@
 from ..trace_utils import unwrap as _u
 
 
+log = get_logger(__name__)
+
```
Comment (mabdinur) [inline ddtrace/contrib/loguru/patch.py:19]:
```suggestion
```

File: ddtrace/contrib/loguru/patch.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import ddtrace
 from ddtrace import config
 
+from ...internal.logger import get_logger
```
Comment (mabdinur) [inline ddtrace/contrib/loguru/patch.py:6]:
```suggestion
```

**Review by zarirhamza — APPROVED** [review]
LGTM! Good catch by wrapping the `configure` method to prevent the patcher being overwritten in the future

File: ddtrace/contrib/loguru/patch.py [hunk 3]
```diff
@@ -47,9 +47,18 @@ def _tracer_injection(event_dict):
...
 
 
-def _w_add(func, instance, args, kwargs):
-    # patch logger to include datadog info before logging
-    instance.configure(patcher=lambda record: record.update(_tracer_injection(record["extra"])))
+def _w_configure(func, instance, args, kwargs):
+    original_patcher = kwargs.get("patcher", None)
+    instance._dd_original_patcher = original_patcher
+    if not original_patcher:
+        # no patcher, we not need to worry about ddtrace fields being overridden
```
Comment (ZStriker19) [inline ddtrace/contrib/loguru/patch.py:54]:
```suggestion
        # no patcher, we do not need to worry about ddtrace fields being overridden
```

**Review by ZStriker19 — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**